internal/validation/types.go

```diff
@@ -31,25 +31,26 @@ func NewValidationError(msg string, errors validator.ValidationErrors) *types.Re
 	}
 }
 
-func buildErrorData(errors validator.ValidationErrors) map[string][]string {
+func buildErrorData(errors validator.ValidationErrors) []string {
 	if errors == nil {
 		return nil
 	}
 
-	failedValidations := make(map[string][]string)
-	for _, e := range errors {
+	failedValidations := make([]string, len(errors))
```

Comment:
issue: the payload validation with specific errors was done for the UI to consume so the user could be presented with field-specific validation errors to act upon.
Switching from a map to an array would invalidate this benefit, making the validation errors useless, since the UI would not be able to grab field-specific errors.
I suggest you stick to the map structure